waiter/src/waiter/service.clj

```diff
@@ -304,5 +304,6 @@
   [deployment-error {:keys [healthy requested scheduled] :or {healthy 0 requested 0 scheduled 0}}]
   (cond
     deployment-error :service-state-failing
+    (and (zero? requested) (zero? scheduled) (zero? healthy)) :service-state-inactive
```

Comment:
Should this simply be `(zero? requested)`?  When would `scheduled` or `healthy` be positive when `requested` is zero?  We should also fix the doc for this fn.

Reply:
Don't we 404 when asked about a service that has no requested instances?  And remove it from the service listing?

Reply:
There can be a gap between requested and actually scheduled instances (delay in killing or scaling up), so we need to check for both values.

We 404 only when the service description is not known. For deleted services, we render the page with empty instances. https://github.com/twosigma/waiter/blob/master/waiter/src/waiter/handler.clj#L442